fairroot/base/steer/FairRunAna.cxx

```diff
@@ -420,13 +411,9 @@ void FairRunAna::RunEventReco(Int_t Ev_start, Int_t Ev_end)
          * if we have simulation files then they have MC Event Header and the Run Id is in it, any way it
          * would be better to make FairMCEventHeader a subclass of FairEvtHeader.
          */
-        if (tmpId != fRunId) {
-            fRunId = tmpId;
-            if (!fStatic) {
-                Reinit(fRunId);
-                fTask->ReInitTask();
-            }
-        }
+
+        CheckRunIdChanged(0);
+
```

Comment:
This is quite confusing to me!

But this is, what the old code did (at least from my reading of it).

Can someone please comment on this?

Reply:
This is a clear mistake, it should simply `CheckRunIdChanged()`.

The `FairRunAna::RunEventReco()` is not being tested nor used by any experiment.
We should most likely deprecate it together with `fairroot/base/event/FairEventBuilder*`

Reply:
I will fix the `CheckRunIdChanged` in a force-psuh soon then.

It looks like PandaRoot uses some of the functions/classes you suggest to deprecate?

Reply:
From what I can see, it's only the GEM dummy implementation that was introduced by me in PandaRoot uses the code.
It did not catch up, so I guess it's not really needed.

Reply:
I am sure, you know better. :-)

Would you mind creating the PR for the deprecation? So that we can handle that independently of this PR?